clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -756,6 +756,11 @@ protected Node curNode() {
             return curNode;
         }
 
+        void abortAndFail(Throwable throwable) {
```

Comment:
How about changing the input type from `Throwable` to `TimeoutException`? The exception passed to this method should NOT be wrapped by `TimeoutException`. Using explicit type (`TimeoutException`) can prevent us from passing incorrect exception in the future.

Reply:
Makes sense.